### DISCUSSION TITLE
🐰 Who is using Keploy? 🧡 [Early Adopters]

### DISCUSSION THREAD

**Discussion by nehagup** [keploy/keploy#1765] [Show and tell]
Sharing experiences and learning from other users is essential. We are frequently asked who is using a particular feature of Keploy so people can get in contact with other users to share experiences and best practices. People also often want to know if product/platform X has integrated Keploy.

The following is a directory of early adopters to help identify users of individual features. The users themselves directly maintain the list.

**Adding yourself as an early adopter**
If you're using Keploy or like the project, please share your comments on this GitHub discussion thread (tell us your Name, Use-Case, Org, Since when, feedback..etc). Your feedback helps us improve!  💚

**Comment by AnkitKumarvaid** [comment]
As a lead product manager of Plypicker, I can confidently say that Keploy has revolutionized our e-commerce efficiency. With its seamless integration and effortless test creation, Keploy ensures our backend remains robust and aligned with customer needs. The reduction in redundancy through test deduplication has saved us valuable development time, while the flawless integration with our existing frameworks ensures a smooth transition. Thanks to Keploy, Plypicker can guarantee a reliable backend that supports our mission of providing a frictionless shopping experience for our customers.

**Comment by aerowisca** [comment]
I'm Mahfooz, an undergraduate student from IIT Dhanbad. Although I do not have a commercial use-case, I've been using Keploy for my personal projects for over 1.5 years now, with the most sophisticated example being a web app that acts as an online judge and code executioner. To perform these operations, it has to retrieve code from external sources. Before Keploy, we had extensive mocks in place to make sure that the automation tests succeeded (for deterministic behaviour, it's not feasible to query websites directly during E2E tests). This was workflow had several pitfalls and maintenance cost with the ever changing source code structure on websites. Keploy was able to transition it to a much smoother workflow by minimizing the manul efforts needed for creating the mocks and providing coverage for the source code. It's not perfect, but I've been closely following the recent developments, and Keploy has come a long way since its early release. I especially like the user friendly approach and the fast onboarding time for new users.

**Comment by anirudhjain75** [comment]
As a senior engineer at Signzy.  I've been using keploy for quite a bit. Getting started was a bit of an hiccup but was easily resolved by the team at Keploy. I've been using it for projects with jest on node along with mongo and redis as dependencies. We are able to capture tests and mocks as if they were real traffic and replay them for anomalies. Its reduces manual effort and provides you great coverage as well. Making the onboarding as simple and easier as possible would be great.

**Comment by himanshu-wedensday** [comment]
As a senior software engineer at Wednesday Solutions, I've extensive experience using Keploy, which has significantly streamlined our development processes. Initially, there was a slight learning curve, but the Keploy team promptly assisted us in overcoming any hurdles.

We primarily use Keploy in projects involving Golang with Postgres and Redis as dependencies, along with GoTest for testing. This combination allows us to capture and replay tests and mocks resembling real traffic, enabling us to detect anomalies effectively. This automation not only reduces manual effort but also enhances our test coverage remarkably.

**Comment by 0xquark** [comment]
As someone who is developing and maintaining open source software for organizations, I mainly use Keploy alongside Golang and GoTest for testing purposes. This combination allows me to record and replay traffic, essentially mimicking how different parts of the system interact with each other. It helps me identify issues more efficiently as well as increasing test coverage. 
On a personal note, I've been using Keploy for over 2 months, primarily for a personal web application. Before Keploy, I heavily relied on manually created unit tests, which became burdensome due to the frequent changes in external source code structures. Keploy has streamlined my workflow by reducing the need for manual testing and providing a more comprehensive coverage of the codebase.
Keploy is ❤️

**Comment by Akshaykumar05** [comment]
Hello folks, I'm Akshay Kumar, currently working as a DevOps Engineer at National Informatics Centre and I've recently started using Keploy in some of my side projects, and I can't comprehend the fact that it has been extremely helpful for me.

I was able to get started with Keploy really easily because of the awesome and detailed documentation. I have used it for projects with Django along with Postgres, and I was able to capture the tests and generate mocks seamlessly. It significantly helped me as writing test cases manually is such a tedious job for me. Also, the fact that I can also generate the test coverage of the unit tests for my application using Keploy is a huge bonus!!

**Comment by codexrahulKIIT** [comment]
Hello everyone I’m Rahul and I’m doing my MCA from KIIT. I just started my open-source adventure as part of the Keploy API Fellowship.
This was my introduction to GitHub PRs, and I learned about contributions. All credits to Keploy & @amaan-bhati and Team for giving such opportunity for begginners like me and guiding us. Excited to be able to give back more and learn with the community!

**Reply by amaan-bhati** [reply]
Hey @codexrahulKIIT We are glad that you liked our initiative. We hope this would help you bring the best out of your potential and you'll be able to contribute to open source and help others in the community!
